spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/GatewayApiController.java

```diff
@@ -51,4 +69,28 @@ public ApiKey findApiKeyEntityByApiKeyValue(@RequestParam("keyValue") String api
     public ServiceApi findServiceByApiKeyValue(@RequestParam("apiKeyValue") String apiKeyValue) {
         return serviceApiRepository.findByApiKeyValue(apiKeyValue);
     }
+
+    //Due to legacy code reasons the gateway API key and the gateway id are exactly the same.
+    @GetMapping("/api/gateways/api-keys")
+    public ResponseEntity<List<ApiKey>> findApiKeyListByGatewayApiKey(@RequestParam("gateway-api-key") String gatewayIdAndApiKey) {
+        log.debug("The gateway with the ID {} requests its latest API key list.", gatewayIdAndApiKey);
+        try {
+            Proxy proxy = proxyRepository.findById(gatewayIdAndApiKey);
+            return new ResponseEntity<>(getApiKeysBelongingToServicesOf(proxy), HttpStatus.OK);
+        } catch (Exception e) {
+            log.info("The creation of the API key list for the gateway {} failed.", gatewayIdAndApiKey, e);
```

Comment:
If something "failed due to an unexpected exception", we should use `log.warn` or `log.error`.

Reply:
Done.